source/tomopy/misc/corr.py

```diff
@@ -598,6 +598,9 @@ def remove_ring(rec, center_x=None, center_y=None, thresh=300.0,
     else:
         raise ValueError("int_mode should be WRAP or REFLECT")
 
+    if not 0 <= theta_min <= 360:
+        raise ValueError("theta_min should be in the range 0 - 360")
+
```

Comment:
My understanding is that `theta_min` in the python API is the same a `angular_min` in the C API. Therefore, `theta_min` determines `m_azi`. I agree that it is sensible to check that `theta_min` must be <= zero and < 360 because a ring artifact cannot be longer than 360 degrees. However, is `theta_min` = 360 valid?

Reply:
Since `m_azi` is actually the *radius* of the averaging kernel and not the width (the docs should be fixed to clarify). The limit of `theta_min` should actually be strictly less than 180. We can prove this by looking at the code which assumes the following:
```
kernel_rad + kernel_rad + 1 <= height  // pol_height is int height; int kernel_rad is int m_azi
m_azi = ceil((float) pol_height / 360.0) * angular_min;
```
which may be combined to show that `angular_min < 180`.

Reply:
```suggestion
    if not 0 <= theta_min < 180:
        raise ValueError("theta_min should be in the range [0, 180)")

```

Reply:
I agree that 180 is a better limit, especially if in practice much smaller values are generally used.